util/delete_scheduler.cc

```diff
@@ -286,6 +290,18 @@ Status DeleteScheduler::DeleteTrashFile(const std::string& path_in_trash,
 
     if (need_full_delete) {
       s = env_->DeleteFile(path_in_trash);
+      if (!dir_to_sync.empty()) {
+        std::unique_ptr<Directory> dir_obj;
+        if (s.ok()) {
+          s = env_->NewDirectory(dir_to_sync, &dir_obj);
```

Comment:
just curious why do we need this Directory object?

Reply:
I don't know. I guess it was created to hold an fd of the directory. While DB is opened, this fd is always held and an fsync() to the directory won't need to call open() and close() again.

In this new piece of code I added here, we do call open() and close(), and hope we are OK with it.